demos/rag_agentic/notebooks/level4_agentic_and_mcp.ipynb

```diff
@@ -0,0 +1,289 @@
+      "\n",
+      "\u001b[0m\u001b[33mIf\u001b[0m\u001b[33m you\u001b[0m\u001b[33m're\u001b[0m\u001b[33m interested\u001b[0m\u001b[33m in\u001b[0m\u001b[33m learning\u001b[0m\u001b[33m more\u001b[0m\u001b[33m about\u001b[0m\u001b[33m the\u001b[0m\u001b[33m new\u001b[0m\u001b[33m features\u001b[0m\u001b[33m and\u001b[0m\u001b[33m capabilities\u001b[0m\u001b[33m in\u001b[0m\u001b[33m Open\u001b[0m\u001b[33mShift\u001b[0m\u001b[33m \u001b[0m\u001b[33m4\u001b[0m\u001b[33m.\u001b[0m\u001b[33m18\u001b[0m\u001b[33m,\u001b[0m\u001b[33m I\u001b[0m\u001b[33m recommend\u001b[0m\u001b[33m checking\u001b[0m\u001b[33m out\u001b[0m\u001b[33m the\u001b[0m\u001b[33m official\u001b[0m\u001b[33m Red\u001b[0m\u001b[33m Hat\u001b[0m\u001b[33m website\u001b[0m\u001b[33m for\u001b[0m\u001b[33m more\u001b[0m\u001b[33m information\u001b[0m\u001b[33m.\n",
+      "\n",
+      "\u001b[0m\u001b[33mBest\u001b[0m\u001b[33m,\n",
+      "\u001b[0m\u001b[33m[\u001b[0m\u001b[33mYour\u001b[0m\u001b[33m Name\u001b[0m\u001b[33m]\n",
+      "\n",
+      "\u001b[0m\u001b[33mSent\u001b[0m\u001b[33m from\u001b[0m\u001b[33m my\u001b[0m\u001b[33m llama\u001b[0m\u001b[33m-stack\u001b[0m\u001b[33m agent\u001b[0m\u001b[97m\u001b[0m\n",
+      "\u001b[30m\u001b[0m"
+     ]
+    }
+   ],
+   "source": [
+    "from llama_stack_client.lib.agents.agent import Agent\n",
+    "\n",
+    "agent = Agent(\n",
+    "    client=client,\n",
+    "    model=model,\n",
+    "    instructions=\"\"\"You are a helpful AI assistant, responsible for helping me find and communicate information back to my team.\n",
+    "    You have access to a number of tools.\n",
+    "    Whenever a tool is called, be sure return the Response in a friendly and helpful tone.\n",
+    "    When you are asked to search the web you must use a tool.\n",
+    "    When signing off on emails, please be sure to include: - Sent from my llama-stack agent in the signature\n",
+    "    \"\"\",\n",
+    "    tools=[\"builtin::websearch\", \"mcp::openshift\"],\n",
+    "    tool_config={\"tool_choice\":\"auto\"},\n",
+    "    sampling_params={\n",
+    "        \"max_tokens\":4096,\n",
+    "        \"strategy\": {\"type\": \"greedy\"},\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "\n",
+    "session_id = agent.create_session(session_name=\"Draft_email_with_latest_OCP_version\")\n",
+    "prompt = \"\"\"Search for the web for the latest Red Hat OpenShift version on the Red Hat website. Summarize the version number and draft an email to convey this information.\"\"\"\n",
+    "turn_response = agent.create_turn(\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\":\"user\",\n",
+    "            \"content\": prompt\n",
+    "        }\n",
+    "    ],\n",
+    "    session_id=session_id,\n",
+    "    stream=True,\n",
+    ")\n",
+    "for log in EventLogger().log(turn_response):\n",
+    "    log.print()\n"
+   ]
+  },
+  {
```

Comment:
Can you also add a few lines to briefly explain the output of this query?

Reply:
done.